styx-scheduler-service/src/main/java/com/spotify/styx/BackfillTriggerManager.java

```diff
@@ -168,7 +168,7 @@ private void triggerAndProgress0(Backfill backfill) {
     final Instant initialNextTrigger = backfill.nextTrigger();
     while (true) {
       try {
-        if (!storage.runInTransaction(tx ->
+        if (!storage.runInTransactionWithRetries(tx ->
```

Comment:
Not sure of this retry either. It should be OK to give up on this backfill and move to the next one.

Reply:
Same. I don't believe there is much benefit in having a non-retrying flavor of `runInTransaction`.
Consider the rationale for the google provided `Datastore.runInTransaction` method retrying conflicts by default. This is a common pattern in transaction storage engine clients.

Reply:
I see your point, and I think there is a difference between client level and application level retry because at application level we encode logic not only plain CRUD. We might need to be more cautious when doing retry.

Reply:
I'm more worried about this transaction code is not idempotent, but maybe it is.

Reply:
After having worked with this code a bit, I now believe its behavior to be correct, although not consistent if there are concurrent updates.

I have refactored the backfill trigger manager code in PR #744 in order to (hopefully) make it easier to understand and give more consistent behavior when there are concurrent updates.